ds3-sdk-integration/src/test/java/com/spectralogic/ds3client/integration/CustomParser_Test.java

```diff
@@ -0,0 +1,158 @@
+            }
+        });
+    }
+
+    @After
+    public void afterRunningTestMethod() throws IOException {
+        deleteAllContents(client, BUCKET_NAME);
+    }
+
+    @Test
+    public void getObjectCustomParser_Test() throws IOException, URISyntaxException {
+        final Ds3Object object = new Ds3Object(OBJ_NAME);
+        final GetBulkJobSpectraS3Response getBulkJobSpectraS3Response = client
+                .getBulkJobSpectraS3(new GetBulkJobSpectraS3Request(BUCKET_NAME, Lists.newArrayList(object)));
+
+        final ByteArrayOutputStream stream = new ByteArrayOutputStream();
+        final GetObjectRequest request = new GetObjectRequest(
+                BUCKET_NAME,
+                OBJ_NAME,
+                getBulkJobSpectraS3Response.getMasterObjectList().getJobId(),
+                0,
+                stream);
+
+        client.getObject(request, new Function<GetObjectCustomParserParameters, GetObjectResponse>() {
+            @Override
+            public GetObjectResponse apply(final GetObjectCustomParserParameters getObjectParserParams) {
+                final WebResponse webResponse = getObjectParserParams.getWebResponse();
+                final  long objectSize = getSizeFromHeaders(webResponse.getHeaders());
+                try (final InputStream responseStream = webResponse.getResponseStream()) {
+                    final long totalBytes = IOUtils.copy(
+                            responseStream,
+                            getObjectParserParams.getDestinationChannel(),
+                            getObjectParserParams.getBufferSize(),
+                            getObjectParserParams.getObjectName(),
+                            false);
+                    getObjectParserParams.getDestinationChannel().close();
+
+                    if (objectSize != -1 && totalBytes != objectSize) {
+                        throw new ContentLengthNotMatchException(getObjectParserParams.getObjectName(), objectSize, totalBytes);
+                    }
+
+                } catch (final IOException e) {
+                    fail();
+                }
+                return null;
+            }
+        });
+        assertThat(stream.toByteArray(), is(OBJ_BYTES));
+    }
+
+    @SuppressWarnings("deprecation")
+    @Test (expected = FailedRequestException.class)
+    public void getObjectCustomParserNoObject_Test() throws IOException{
+        final String objectName = "doesNotExistObject";
+
+        final GetObjectRequest request = new GetObjectRequest(
+                BUCKET_NAME,
+                objectName,
+                new NullChannel());
+
+        //Get object that does not exist
+        client.getObject(request, new Function<GetObjectCustomParserParameters, GetObjectResponse>() {
+            @Override
+            public GetObjectResponse apply(final GetObjectCustomParserParameters getObjectParserParams) {
+                fail();
+                return null;
+            }
+        });
+        fail();
```

Comment:
You don't need this fail check since you have the `@Test (expected = ...)` annotation